protocol/auditlog.go

```diff
@@ -81,24 +109,21 @@ func (l ConiksAuditLog) get(dirInitHash [crypto.HashSizeByte]byte) (*directoryHi
 	return h, ok
 }
 
-// Insert creates a new directory history for the key directory addr
+// InitHistory creates a new directory history for the key directory addr
 // and inserts it into the audit log l.
 // The directory history is initialized with the key directory's
-// signing key signKey, and a list of snapshots snaps representing the
-// directory's STR history so far, in chronological order.
-// Insert() returns an ErrAuditLog if the auditor attempts to create
+// signing key signKey, and a list of one or more snapshots snaps
+// representing the pinned directory's STR history so far,
+// in chronological order.
+// InitHistory() returns an ErrAuditLog if the auditor attempts to create
 // a new history for a known directory, and nil otherwise.
-// Insert() only creates the initial entry in the log for addr. Use Update()
-// to insert newly observed STRs for addr in subsequent epochs.
-// Insert() assumes that the caller
-// has called Audit() on snaps before calling Insert().
-// FIXME: pass Response message as param
-// masomel: will probably want to write a more generic function
-// for "catching up" on a history in case an auditor misses epochs
-func (l ConiksAuditLog) Insert(addr string, signKey sign.PublicKey,
+// InitHistory() is called by an auditor when it initializes its state
+// from disk (either first-time startup, or after reboot).
+func (l ConiksAuditLog) InitHistory(addr string, signKey sign.PublicKey,
 	snaps []*DirSTR) error {
 	// make sure we're getting an initial STR at the very least
 	if len(snaps) < 1 || snaps[0].Epoch != 0 {
```

Comment:
I think we should compare `snaps[0]` with the pinning STR, no?

Reply:
`snaps[0]` *is* the pinned STR. As the documentation describes, this function is called when the auditor first starts up, so the very first time that auditor is launched, it will read the pinned STR from disk and pass it into `InitHistory` through the `snaps` parameter.

Reply:
Ahh sorry, I missed that part. Thanks!

Reply:
No problem! I can make this clearer in the documentation.